pyqldb/driver/qldb_driver.py

```diff
@@ -279,71 +296,63 @@ def _create_new_session(self):
         Create a new QldbSession object.
         """
         session_client = SessionClient._start_session(self._ledger_name, self._client)
-        return QldbSession(session_client, self._read_ahead, self._executor, self._release_session)
+        return QldbSession(session_client, self._read_ahead, self._executor)
 
     def _release_session(self, session):
         """
         Release a session back into the pool.
         """
+        if session:
+            self._pool_permits.release()
+            self._pool_permits_counter.increment()
+            logger.debug('Number of sessions in pool : {}'.format(self._pool.qsize()))
```

Comment:
This debug message should only be printed and be after putting a session back in the pool.

Reply:
Moved